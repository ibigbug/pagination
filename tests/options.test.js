import { mount } from 'enzyme';
import Select from 'rc-select';
import zhCN from '../src/locale/zh_CN';
import Options from '../src/Options';

const WrapperOptions = (props) => {
  return (
    <Options
      locale={zhCN}
      rootPrefixCls="rc-pagination"
      selectComponentClass={Select}
      pageSize={10}
      changeSize={jest.fn()}
      quickGo={jest.fn()}
      {...props}
    />
  );
};

describe('Options', () => {
  it('should render correctly', () => {
    const wrapper = mount(<WrapperOptions />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('props:buildOptionText', () => {
    it('should render correctly', () => {
      const mockBuildOptionText = jest
        .fn()
        .mockImplementation((value) => (
          <div className="custom-options">buildOptionText-{value}</div>
        ));
      const wrapper = mount(
        <WrapperOptions buildOptionText={mockBuildOptionText} />,
      );
      const options = wrapper.find('.custom-options');
      expect(options).toBeTruthy();
      expect(options.text()).toBe('buildOptionText-10');
    });
  });
});
