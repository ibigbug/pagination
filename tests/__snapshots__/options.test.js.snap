// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should render correctly 1`] = `"<li class="rc-pagination-options"><div class="rc-select rc-pagination-options-size-changer rc-select-single" aria-label="页码"><div class="rc-select-selector"><span class="rc-select-selection-search"><input type="search" autocomplete="off" class="rc-select-selection-search-input" role="combobox" aria-expanded="false" aria-haspopup="listbox" aria-owns="rc_select_TEST_OR_SSR_list" aria-autocomplete="list" aria-controls="rc_select_TEST_OR_SSR_list" aria-label="页码" readonly="" unselectable="on" style="opacity: 0;" value="" id="rc_select_TEST_OR_SSR"></span><span class="rc-select-selection-item" title="10 条/页">10 条/页</span></div></div><div class="rc-pagination-options-quick-jumper">跳至<input type="text" aria-label="页" value="">页</div></li>"`;
